vllm/model_executor/model_loader/loader.py

```diff
@@ -526,6 +533,201 @@ def save_model(
             )
 
 
+class BitsAndBytesModelLoader(BaseModelLoader):
+    """Model loader that will set model weights to random values."""
+
+    def __init__(self, load_config: LoadConfig):
+        super().__init__(load_config)
+        default_target_modules = [
+            "gate_proj", "down_proj", "up_proj", "q_proj", "k_proj", "v_proj",
+            "o_proj"
+        ]
```

Comment:
let's make it a class attribute. also isn't it better to always just read from the config? is this a common way to handle unspecified layers? not familiar with this myself!

Reply:
BnB can work without a LoRA adpater. Such an example was given in hugging face blog, https://huggingface.co/blog/4bit-transformers-bitsandbytes. 

What i am doing here is following the handling of BitsAndBytesConfig for Llama model in hugging face transformer package. Namely, only the listed modules will be quantized, while other modules, such as lm_head, embed_tokens, layer_norms, are not quantized. 

Since no quant_config files to read if there is not LoRA adapter specified, I have to put them as hard-coded. 

However, I have to say, the list of default target modules may be different for other models. But since only Llama is supported in this MR, I just put it as is. I will change if necessary, that is for sure.